--- ffmpeg/cmdutils.c  2012-10-08 09:14:33.928824569 +0100
+++ ffmpeg/cmdutils.c  2012-10-08 09:16:57.292868973 +0100
@@ -608,7 +608,7 @@
 
 #define INDENT        1
 #define SHOW_VERSION  2
-#define SHOW_CONFIG   4
+#define SHOW_CONFIG   0 /* disable */
 #define SHOW_COPYRIGHT 8
 
 #define PRINT_LIB_INFO(libname, LIBNAME, flags, level)                  \
@@ -654,11 +654,13 @@
 #endif
 }
 
+#define FFMPEG_CONFIGURATION "http://git.io/ZHdseg"
+#define PMS_FFMPEG_VERSION 4
 static void print_program_info(int flags, int level)
 {
     const char *indent = flags & INDENT? "  " : "";
 
-    av_log(NULL, level, "%s version " FFMPEG_VERSION, program_name);
+    av_log(NULL, level, "%s version %s (PMS%d) for PS3 Media Server", program_name, FFMPEG_VERSION, PMS_FFMPEG_VERSION);
     if (flags & SHOW_COPYRIGHT)
         av_log(NULL, level, " Copyright (c) %d-%d the FFmpeg developers",
                program_birth_year, this_year);
@@ -676,6 +678,7 @@
         return;
 
     print_program_info (INDENT|SHOW_COPYRIGHT, AV_LOG_INFO);
+    return;
     print_all_libs_info(INDENT|SHOW_CONFIG,  AV_LOG_INFO);
     print_all_libs_info(INDENT|SHOW_VERSION, AV_LOG_INFO);
 }
